jetbrains-core/src/software/aws/toolkits/jetbrains/ui/wizard/SamInitProjectBuilderCommon.kt

```diff
@@ -0,0 +1,81 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.ui.wizard
+
+import com.intellij.ide.util.projectWizard.ModuleBuilder
+import com.intellij.openapi.module.ModuleType
+import com.intellij.openapi.options.ShowSettingsUtil
+import com.intellij.openapi.project.DefaultProjectFactory
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.platform.ProjectTemplate
+import icons.AwsIcons
+import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.services.lambda.execution.sam.SamInitRunner
+import software.aws.toolkits.jetbrains.settings.AwsSettingsConfigurable
+import software.aws.toolkits.jetbrains.settings.SamSettings
+import software.aws.toolkits.jetbrains.ui.wizard.java.SamInitModuleBuilder
+import software.aws.toolkits.resources.message
+import javax.swing.JButton
+import javax.swing.JTextField
+
+class SamProjectTemplateWrapper(
+    val samProjectTemplate: SamProjectTemplate,
+    val builder: ModuleBuilder
+) : ProjectTemplate {
+    override fun getIcon() = samProjectTemplate.getIcon()
+
+    override fun getName() = samProjectTemplate.getName()
+
+    override fun getDescription() = samProjectTemplate.getDescription()
+
+    override fun createModuleBuilder() = builder
+
+    override fun validateSettings() = null
+}
+
+abstract class SamProjectTemplate {
```

Comment:
Long term I'd expect `SamProjectTemplate` to have a runtime associated with it too (or maybe a runtime group).